specification/communication/resource-manager/Microsoft.Communication/preview/2023-03-01-preview/SenderUsernames.json

```diff
@@ -0,0 +1,333 @@
+          },
+          {
+            "$ref": "EmailServices.json#/parameters/EmailServiceName"
+          },
+          {
+            "$ref": "Domains.json#/parameters/DomainName"
+          },
+          {
+            "$ref": "#/parameters/SenderUsername"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The delete operation was successful."
+          },
+          "404": {
+            "description": "The resource did not exist."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Delete SenderUsernames resource": {
+            "$ref": "./examples/senderUsernames/delete.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "SenderUsernameResource": {
+      "description": "A class representing a SenderUsername resource.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/SenderUsernameProperties",
+          "description": "The properties of a SenderUsername resource.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "SenderUsernameResourceCollection": {
+      "description": "A class representing a Domains SenderUsernames collection.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "List of SenderUsernames",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SenderUsernameResource"
+          }
+        },
+        "nextLink": {
+          "description": "The URL the client should use to fetch the next page (per server side paging).\r\nIt's null for now, added for future use.",
+          "type": "string"
+        }
+      }
+    },
+    "SenderUsernameProperties": {
+      "description": "A class that describes the properties of a SenderUsername resource.",
+      "type": "object",
+      "required": [
+        "username"
+      ],
+      "properties": {
+        "dataLocation": {
+          "description": "The location where the SenderUsername resource data is stored at rest.",
+          "type": "string",
+          "readOnly": true
+        },
+        "username": {
+          "description": "A sender senderUsername to be used when sending emails.",
+          "type": "string"
+        },
+        "displayName": {
+          "description": "The display name for the senderUsername.",
+          "type": "string"
+        },
+        "provisioningState": {
+          "description": "Provisioning state of the resource.",
+          "enum": [
+            "Unknown",
```

Comment:
>Unknown [](http://example.com/codeflow?start=13&length=7)

Is Unknown a real state or is it just for representing future unknown states?

Reply:
PS: consider adding a description of the enum value.

Reply:
I have updated the description of the enum.